yuansfer/api/base_api.py

```diff
@@ -19,35 +20,22 @@ def __init__(self, config):
     def config(self):
         return self._config
 
-    def date_validate(self, name, value):
-        # Check if date exists
-        if value is None:
-            raise RequireParamsError(params=name)
-
-        reg = r"^\\d{4}\\d{2}\\d{2}$"
-
-        if not re.match(reg, value):
-            raise InvalidParamsError("data error: " + name)
-
-    def number_validate(self, name, value):
-        # Check if number exists
+    def validation(self,param, value, name):
+        # Check if params exist
         if value is None:
-            raise RequireParamsError(params=name)
+            raise RequireParamsError(params= param)
```

Comment:
should it be `params=param` ?

Reply:
yes, the param is the name of the field for the request body
The "name" param should be the validationType, i am changing it now